src/cargo/sources/git/source.rs

```diff
@@ -169,7 +169,10 @@ impl<'cfg> Source for GitSource<'cfg> {
                 format!("git repository `{}`", self.remote.url())));
 
             trace!("updating git source `{:?}`", self.remote);
-            let repo = try!(self.remote.checkout(&self.db_path));
+
+            let repo = try!(network::with_retry(&self.config, ||{
+              self.remote.checkout(&self.db_path)
```

Comment:
Ah it should be fine to probably just pass in the Config instead of storing it. Each source already stores a pointer to `Config` somewhere so it should be readily available to pass in